### DISCUSSION TITLE
TraceContext will lost in SpringBootTest junit test

### DISCUSSION THREAD

**Discussion by taotao365s** [apache/skywalking#9496] [Q&A] [answered]
when I debug in SpringBootTest test，if I debug code，the trace id will lost。for example，

before all, javaagent option is configured, and it is ok

````
@SpringBootTest
public class BaseSpringTest {

  @Autowired
  protected UserContext userContext;

  @Trace
  public void testOne() {
    log.info("before each test");
     log.info("test insert"); // debug breakpoint
    log.info("test insert2"); // debug breakpoint

// do some business
// then get value from trace context
// TraceContext.getColouration("xxx-accountId"); // here is NULL
  }

````

**Comment by wu-sheng** [comment, marked as answer]
SkyWalking Java agent is not designed for short life cycle applications, so it doesn't work for this kind of UT too. 

First line of project intro **SkyWalking is an open source APM system, including monitoring, tracing, diagnosing capabilities for distributed system in Cloud Native architecture**
